flake.nix

```diff
@@ -1,14 +1,19 @@
 {
-  outputs = { self, nixpkgs }: {
-    lib = {
-      sources = import ./sources.nix { inherit (nixpkgs) lib; };
-
-      # this does lock us to a different nixpkgs version for the docker build
-      # steps but it ensures the API is what we expect
-      docker = import ./docker.nix {
-        pkgs = import nixpkgs { system = "x86_64-linux"; };
-      };
+  inputs.flake-utils.url = "github:numtide/flake-utils";
+  outputs = { self, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = import nixpkgs { inherit system; };
+        inherit (pkgs) callPackage;
+      in
+      {
+        # this does lock us to a different nixpkgs version for the docker build
+        # steps but it ensures the API is what we expect
```

Comment:
I think you can just remove this comment now. It's technically correct but it depends on how you use the flakes. A consumer can override what nixpkgs the flake will use so it's a bit moot.